internal/controller/numaresourcesoperator_controller.go

```diff
@@ -780,21 +781,18 @@ func daemonsetUpdater(poolName string, gdm *rtestate.GeneratedDesiredManifest) e
 	// We cannot do this at GetManifests time because we need to mount
 	// a specific configmap for each daemonset, whose name we know only
 	// when we instantiate the daemonset from the MCP.
-	if gdm.ClusterPlatform != platform.OpenShift && gdm.ClusterPlatform != platform.HyperShift {
-		klog.V(5).InfoS("DaemonSet update: unsupported platform", "pool name", poolName, "platform", gdm.ClusterPlatform)
-		// nothing to do!
-		return nil
-	}
-	err = rteupdate.ContainerConfig(gdm.DaemonSet, gdm.DaemonSet.Name)
-	if err != nil {
-		// intentionally info because we want to keep going
-		klog.V(5).InfoS("DaemonSet update: cannot update config", "pool name", poolName, "daemonset", gdm.DaemonSet.Name, "error", err)
-		return err
-	}
-	if gdm.ClusterPlatform != platform.Kubernetes {
+	if gdm.ClusterPlatform == platform.OpenShift || gdm.ClusterPlatform == platform.HyperShift {
+		klog.V(4).InfoS("DaemonSet update: container configuration", "platform", gdm.ClusterPlatform, "daemonSetName", gdm.DaemonSet.Name)
+		err = rteupdate.ContainerConfig(gdm.DaemonSet, gdm.DaemonSet.Name)
+		if err != nil {
+			// intentionally info because we want to keep going
+			klog.V(5).InfoS("DaemonSet update: cannot update config", "pool name", poolName, "daemonset", gdm.DaemonSet.Name, "error", err)
+			return err
+		}
```

Comment:
I agree we need to enhance the conditions but I think we should return early on unsupported platform. the problem with this version of code is that `rteupdate.DaemonSetHashAnnotation(gdm.DaemonSet, gdm.RTEConfigHash)` (https://github.com/openshift-kni/numaresources-operator/pull/2532/files#diff-90fa307399f0b81f0128df9422f0108dada141ab749d0995fba26492a68085a5R797) will be executed for unsupported platform, is this done on purpose?